app/controllers/taxon_framework_relationships_controller.rb

```diff
@@ -0,0 +1,203 @@
+class TaxonFrameworkRelationshipsController < ApplicationController
+  before_filter :authenticate_user!, :except => [:index, :show]
+  before_filter :admin_required, :only => [:new, :create, :edit, :update, :destroy]
+  before_action :set_taxon_framework_relationship, except: [:index, :new, :create]
+  
+  layout "bootstrap"
+  
+  def index
+    filter_params = params[:filters] || params
+    @relationships = filter_params[:relationships]
+    @relationships ||= TaxonFrameworkRelationship::RELATIONSHIPS.map{ |r| filter_params[r] == "1" ? r : nil }
+    @relationships.delete_if{ |r| r.blank? }
+    @relationships = @relationships.map{ |r| r.gsub( "_"," " ) }
+    @taxon_framework = TaxonFramework.find_by_id( filter_params[:taxon_framework_id] ) unless filter_params[:taxon_framework_id].blank?    
+    @taxon_frameworks = TaxonFramework.all.limit( 100 )
+    @taxon = Taxon.find_by_id( filter_params[:taxon_id].to_i ) unless filter_params[:taxon_id].blank?
+    user_id = filter_params[:user_id] || params[:user_id]
+    @user = User.find_by_id(user_id) || User.find_by_login(user_id) unless user_id.blank?
+    @is_active = filter_params[:is_active]
+    @rank = filter_params[:rank] unless filter_params[:rank].blank?
+    
+    scope = TaxonFrameworkRelationship.all
+    scope = scope.relationships(@relationships) unless @relationships.blank?
+    scope = scope.taxon_framework(@taxon_framework) if @taxon_framework
+    scope = scope.taxon(@taxon) if @taxon
+    scope = scope.by(@user) if @user
+    scope = scope.rank(@rank) if @rank
+    if @is_active == "True"
```

Comment:
Since deriving boolean values from form inputs can be a bit brittle, we usually try to use the `yesish?` and `noish?` methods, so this would be `if @is_active.yesish?`

Reply:
done